packages/anoncreds/src/formats/LegacyIndyProofFormatService.ts

```diff
@@ -488,32 +488,16 @@ export class LegacyIndyProofFormatService implements ProofFormatService<LegacyIn
    *
    */
   private async getCredentialDefinitions(agentContext: AgentContext, credentialDefinitionIds: Set<string>) {
-    const registryService = agentContext.dependencyManager.resolve(AnonCredsRegistryService)
-
     const credentialDefinitions: { [key: string]: AnonCredsCredentialDefinition } = {}
 
     for (const credentialDefinitionId of credentialDefinitionIds) {
-      if (!isUnqualifiedCredentialDefinitionId(credentialDefinitionId)) {
-        throw new CredoError(`${credentialDefinitionId} is not a valid legacy indy credential definition id`)
-      }
-
-      const credentialDefinitionRegistry = registryService.getRegistryForIdentifier(
-        agentContext,
-        credentialDefinitionId
-      )
-
-      const credentialDefinitionResult = await credentialDefinitionRegistry.getCredentialDefinition(
-        agentContext,
-        credentialDefinitionId
-      )
-
-      if (!credentialDefinitionResult.credentialDefinition) {
-        throw new CredoError(
-          `Credential definition not found for id ${credentialDefinitionId}: ${credentialDefinitionResult.resolutionMetadata.message}`
-        )
+      const credentialDefinitionResult = await fetchCredentialDefinition(agentContext, credentialDefinitionId)
+      if (isUnqualifiedCredentialDefinitionId(credentialDefinitionResult.credentialDefinitionId)) {
+        credentialDefinitions[credentialDefinitionId] = credentialDefinitionResult.credentialDefinition
+      } else {
+        credentialDefinitions[getUnQualifiedDidIndyDid(credentialDefinitionId)] =
+          getUnqualifiedDidIndyCredentialDefinition(credentialDefinitionResult.credentialDefinition)
       }
```

Comment:
This should not support qualifieid id. So we need to get back the 

```
      if (!isUnqualifiedCredentialDefinitionId(credentialDefinitionId)) {
        throw new CredoError(`${credentialDefinitionId} is not a valid legacy indy credential definition id`)
      }
```

code and then we can remove the if / else here. This is `LegacyIndyProofFormatService` which only supports unqualified.

Reply:
how should we store credentials received via the legacyIndyCredentialFormatService?